sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/SchemaPruningSuite.scala

```diff
@@ -668,6 +669,50 @@ abstract class SchemaPruningSuite
     }
   }
 
+  testSchemaPruning("SPARK-51831: Column pruning with exists Join") {
+    withTempPath { dir =>
+      spark.range(100)
+        .withColumn("col1", col("id") + 1)
+        .withColumn("col2", col("id") + 2)
+        .withColumn("col3", col("id") + 3)
+        .withColumn("col4", col("id") + 4)
+        .withColumn("col5", col("id") + 5)
+        .withColumn("col6", col("id") + 6)
+        .withColumn("col7", col("id") + 7)
+        .withColumn("col8", col("id") + 8)
+        .withColumn("col9", col("id") + 9)
```

Comment:
hmm do we need so many columns to demonstrate the issue?

Reply:
no, it can be cleaned.

Reply:
I have updated the UT, so we no longer have so much code.